lib/ecto/query/builder.ex

```diff
@@ -217,6 +217,11 @@ defmodule Ecto.Query.Builder do
     {expr, params_acc}
   end
 
+  def escape({:=, _, _}, _type, _params_acc, _vars, _env) do
+    error! "match operator is  not supported: `=`. " <>
+           "Instead use the comparison operator: `==`"
```

Comment:
Sounds better indeed, it invites the developer to take a moment and think about what they're trying to achieve instead of blindly replacing `=` with `==`.